MAINTENANCE.md

```diff
@@ -1,25 +0,0 @@
-# Maintainer's Handbook
-
-## Make a new release
```

Comment:
Instead of deleting the file, why not describe that we're using github actions to do it, and show a quick link?

Reply:
Good idea, done!